source/extensions/filters/http/kill_request/kill_request_filter.cc

```diff
@@ -0,0 +1,46 @@
+#include "extensions/filters/http/kill_request/kill_request_filter.h"
+
+#include <csignal>
+
+#include "extensions/filters/common/fault/fault_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace KillRequest {
+
+bool KillRequestFilter::isKillRequestEnabled() {
+  uint32_t denominator = 100;
+  if (kill_request_.probability().denominator() ==
+      envoy::type::v3::FractionalPercent::TEN_THOUSAND) {
+    denominator = 10000;
+  } else if (kill_request_.probability().denominator() ==
+             envoy::type::v3::FractionalPercent::MILLION) {
+    denominator = 1000000;
+  }
+
+  return (random_generator_.random() % denominator) < kill_request_.probability().numerator();
```

Comment:
See existing utilities for this, `ProtobufPercentHelper::evaluateFractionalPercent` etc.

Reply:
Thanks Harvey, switched to evaluateFractionalPercent